rotkehlchen/chain/ethereum/modules/cowswap/decoder.py

```diff
@@ -251,5 +249,5 @@ def counterparties(self) -> list[str]:
     def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
         return {ETH_FLOW_ADDRESS: (self._decode_eth_orders,)}
 
-    def post_decoding_rules(self) -> list[tuple[int, Callable]]:
-        return [(0, self._aggregator_post_decoding)]
+    def post_decoding_rules(self) -> dict[str, list[tuple[int, Callable]]]:
+        return {GPV2_SETTLEMENT_ADDRESS: [(0, self._aggregator_post_decoding)]}
```

Comment:
The problem is that for the cases we handle

- balancer: we need to use counterparty because it uses the dsproxy that is a variable address and you can't really filter by address
- cowswap: we don't know it is a cowswap trade until in the post decoding rule we filter by address. Is not possible to use the counterparty as filter as cowswap doesn't appear in the list of counterparties because is not known during the initial decoding

What I've done is to add a address -> counterparty mapping so we can use it